### ISSUE TITLE
Is it possible to retrieve the wallaby worker id from the actual test code?

### ISSUE THREAD

**Issue by joachimboggild** [wallabyjs/public#774]
### Issue description or question
I am doing a large suite of tests that all interact with a database. I have setup and teardown code that creates the database and then drops it afterwards. This means that if I run tests in parallel, they attempt to access the same database.

Currently I solve this by setting  workers: { initial: 1, regular: 1 } in my wallaby config file, but it would be great if I could run tests in parallel to speed up things.

My thought was to use the worker Id in the database name to guarantee that each separate worker gets its own database to work with. But is it possible to retrieve the worker Id from the test code?

Edit: I found another issue, #707 , which mentions the same thing. But I am not always running my tests with Wallaby, so I do not want to put my setup code in the wallaby configuration file, as that will not be hit in those circumstances. That is why I would like to export the worker id somehow, if possible.

Thanks :-)

### Code editor or IDE name and version

WebStorm v2016.2

### OS name and version

OSX

**Comment by ArtemGovorov** [comment]
Depending on how you're planning ti reuse it, you may make the worked id available to any other code by assigning a global variable:
```javascript
 module.exports = function () {
   return {
     ...
     setup: function (wallaby) {
       global.workerId = wallaby.workerId;
     }
   };
 };
```
or create a file where it's used (to set up your DB for example and just pass it there):
```javascript
 module.exports = function () {
   return {
     files: [
       ...
       'setupFile.js'
     ]
     ...
     setup: function (wallaby) {
       require('./setupFile')(wallaby.workerId);
     }
   };
 };
```
and the `setupFile.js`:

```javascript
module.exports = function (workerId) {
  // your initialization logic 
  //that can be reused by other runner (that can pass some worked id)
}
```

Let me know if it addresses your question.

**Comment by joachimboggild** [comment]
That is fine - the global one should work the easiest. ... if it is not overwritten by the other workers? If I have e.g. six different workers running, will they have different global variables? (This might be off-topic, I know...)

**Comment by ArtemGovorov** [comment]
Correct. Each worker has its own separate node process, the `setup` function is invoked in each of those worker processes, so each will have an individual worker id.

**Comment by joachimboggild** [comment]
Thanks very much!

**Comment by joachimboggild** [comment]
.... a tiny hiccup: I can't seem to get the setup function to work. My wallaby.js config file looks like this: 

module.exports = function () {
  //noinspection JSUnusedGlobalSymbols
    return {
        env: {
          type: 'node',
          params: {
            runner: "--harmony"
          }
        },

        setup: function(wallaby) {
            console.log(`******************* workerId: ${wallaby.workerId} ***********************`);
            global.workerId = wallaby.workerId;
        },

        bootstrap: function() {
          var path = require('path');
          require('co-mocha')(require(path.join(path.dirname(process.argv[1]), 'runners/node/mocha@2.1.0/framework/')));
        },

        files: [
          "bl/**/*.js",
          "dal/**/*.js",
          "model/**/*.js",
          "plugins_v3/**/*.js",
          "plugins/**/*.js",
            "utils/**/*.js",
            "config/**/*.js",
            "test/test_config.js",
            "test/test_db.js",
            "test/test_graph.js",
            "test/mocha.config.test.js",
            "app.js",
        ],

        tests: [
            'test/dal/**/*.test.js',
            'test/plugins/**/*.test.js',
            'test/bl/**/*.test.js',
        ],

        testFramework: "mocha",

        debug: true,

        workers: {
          initial: 6,
          regular: 2
        }
    };
};

And I am looking for the comment specifying the worker id in the Wallaby Console, but I cannot see it anywhere. Am I looking in the wrong place?

**Comment by ArtemGovorov** [comment]
You need to merge the `setup` and the `bootstrap` function into one, these are the [alias of the same thing](https://wallabyjs.com/docs/config/bootstrap.html), so I suspect only the `bootstrap` gets called.

```javascript
setup: function(wallaby) {
            console.log(`******************* workerId: ${wallaby.workerId} ***********************`);
            global.workerId = wallaby.workerId;
            var path = require('path');
            require('co-mocha')(require(path.join(path.dirname(process.argv[1]), 'runners/node/mocha@2.1.0/framework/')));
        },
```

**Comment by joachimboggild** [comment]
Ah, thanks. I will try that.

**Comment by joachimboggild** [comment]
How about using console.log from the setup function: Shouldn't its output appear in the wallaby console window (using WebStorm)?

**Comment by joachimboggild** [comment]
Nvm - it started appearing after I merged the two functions. Thanks very much for your help!
